codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -546,6 +581,7 @@ void incrementStreamableBytesForTree(int numBytes) {
          */
         private final class Frame {
             final ByteBuf data;
+            final Http2Headers headers;
```

Comment:
Ok. I've made this change. It had a fairly extensive impact as you might imagine, particularly on the tests but responsibilities are more cleanly separated now. Tests are much cleaner as a result.